src/certificate-authority.ts

```diff
@@ -111,3 +111,43 @@ async function saveCertificateAuthorityCredentials(keypath: string) {
   let key = readFile(keypath, 'utf-8');
   await currentPlatform.writeProtectedFile(rootCAKeyPath, key);
 }
+
+
+function certErrors(): string {
+  try {
+    openssl(`x509 -in ${ rootCACertPath } -noout`);
+    return '';
+  } catch (e) {
+    return e.toString();
+  }
+}
+
+// This function helps to migrate from v1.0.x to >= v1.1.0.
+/**
+ * Smoothly migrate the certificate storage from v1.0.x to >= v1.1.0.
+ * In v1.1.0 there are new options for retrieving the CA cert directly,
+ * to help third-party Node apps trust the root CA.
+ * 
+ * If a v1.0.x cert already exists, then devcert has written it with
+ * platform.writeProtectedFile(), so an unprivileged readFile cannot access it.
+ * Pre-detect and remedy this; it should only happen once per installation.
+ */
+export async function ensureCACertReadable(options: Options = {}): Promise<void> {
+  if (!certErrors()) {
+    return;
+  }
+  /**
+   * on windows, writeProtectedFile left the cert encrypted on *nix, the cert
+   * has no read permissions either way, openssl will fail and that means we
+   * have to fix it
+   */
+  const caFileContents = await currentPlatform.readProtectedFile(rootCACertPath);
+  await currentPlatform.deleteProtectedFile(rootCACertPath);
+  writeFile(rootCACertPath, caFileContents);
+  
+  // double check that we have a live one
+  const remainingErrors = certErrors();
+  if (remainingErrors) {
+    return installCertificateAuthority(options);
+  }
```

Comment:
I may be misreading this
Should this condition be this?
```diff
- if (remainingErrors) {
+ if (!remainingErrors) {
      return installCertificateAuthority(options);
  }
```
from line 119
```ts
return '' // Success returns a falsy value, yeah?
```

Reply:
Oh, nevermind, I know why you're doing this now.  Just kinda popped into my head 😆 Need to recreate the certificate authority if it is invalid.

So, now my question becomes: shouldn't the old domain certificates, the old CA files, and the old CA from the trust stores be removed?  If creating a new CA, then all of that stuff will be invalid.  

I know that the `installCertificateAuthority` routine checks for older versions.... maybe a new function that can be run in various circumstances when it's desirable to wipe out the current `devcert` CA state?

Reply:
I know it would be easier to read if `certErrors(): string` was like `certIsValid(): boolean`, but I did it this way to lay the groundwork for more sophisticated error reporting in the future.

As for removing the old certificates, you're right--and some of those will also be protected. Time to do even more scrubbing!

Reply:
That was all me.  I don't think it's necessarily hard to read; I just had it in my head for some reason that the next reasonable to step to take upon successful conversion was to run the install routine.  Somehow I forgot that the certificate would have already been installed previously if it got to this point 🤷‍♂.  Doubting whether I was reading it right was my subconscious telling me I absolutely was not, I think. 😄 

> I did it this way to lay the groundwork for more sophisticated error reporting in the future.

Out of curiosity, how would you feel about extending this cert checking & scrubbing process to check for an expired CA + renewing it (and its trusts/signed certs)?  I could write up a PR when I have some more free time.

Reply:
I think that sounds great, @Js-Brecht . But yeah, it can be a subsequent PR.

Because of the breaking change in #41, I've had to revamp the installation/uninstallation process, so I've just updated this PR a fair bit. Please take another look!